### PR TITLE
Feat/native stl io

### PR DESCRIPTION
**Pull request by madsendennis**
Native scala implementation to read/write the STL format.
The implementation is ~6 times faster to write a PLY file and ~2.5 times faster to read. 

This is the first part of removing the VTK dependency in Scalismo, https://github.com/unibas-gravis/scalismo/issues/415

### CODE REVIEW

**Review by Andreas-Forster — APPROVED** [review]
I like the changes, they look reasonable small. To my suggestions, check what else need to be changed, especially if you accept the proposed renaming. This is not properly propagated.

As there is a test in place, I did not check every detail of the format how it is written.

File: src/main/scala/scalismo/io/stl/STLMeshWriter.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import scalismo.geometry.EuclideanVector3D
+import scalismo.mesh.TriangleMesh3D
+
+import java.io.{BufferedOutputStream, DataOutputStream, FileOutputStream}
+import java.nio.{ByteBuffer, ByteOrder}
+import scala.util.Try
+
+object STLMeshWriter {
+  private val ORDER = ByteOrder.LITTLE_ENDIAN
```
[the remark this answers is not in the document]
Comment (madsendennis) [reply src/main/scala/scalismo/io/stl/STLMeshWriter.scala:26]: From wikipedia: "numbers and are assumed to be little-endian, although this is not stated in documentation". I can add this comment in the file as well.

File: src/main/scala/scalismo/io/stl/STLMeshReaderAscii.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    breader.close()
+
+    STLHelpers.STLTrianglesToTriangleMesh(triangles.toSeq)
+  }
+
+  private def parseTriangleStrings(data: Seq[String]): Try[STLTriangle] = Try {
+    if (data.length != 7 ||
+      !data(0).startsWith("facet normal") ||
+      !data(1).startsWith("outer loop") ||
+      !data(2).startsWith("vertex") ||
```
[the remark this answers is not in the document]
Comment (madsendennis) [reply src/main/scala/scalismo/io/stl/STLMeshReaderAscii.scala:52]: I'll move this to a separate checkTriangleFormat function

File: src/main/scala/scalismo/io/stl/STLMeshWriter.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      writeVertex(dos, p2)
+      writeVertex(dos, p3)
+      writeShort(dos, 0)
+    }
+    dos.close()
+  }
+
+  private def writeString(dos: DataOutputStream, data: String): Unit = {
+    dos.write(ByteBuffer.allocate(data.getBytes.length)
+      .order(ORDER).put(data.getBytes("UTF-8")).array())
```
[the remark this answers is not in the document]
Comment (madsendennis) [reply src/main/scala/scalismo/io/stl/STLMeshWriter.scala:51]: Good spot. I see that both stl and ply in fact support ASCII, so should be updated accordingly.

File: src/main/scala/scalismo/io/stl/STLMeshReaderAscii.scala [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      parseVertexString(filtered3Ddata(4))
+    )
+  }
+
+  private def parseNormalString(parts: Seq[String]): EuclideanVector3D = {
+      EuclideanVector3D(parts(0).toFloat, parts(1).toFloat, parts(2).toFloat)
+  }
+
+  private def parseVertexString(parts: Seq[String]): Point3D = {
+      Point3D(parts(1).toFloat, parts(1).toFloat, parts(2).toFloat)
```
Comment (Andreas-Forster) [inline src/main/scala/scalismo/io/stl/STLMeshReaderAscii.scala:75]: The indices seem to be a bug here. (1 1 2)

I will propose a PR soon also fixing this bug.

**Review by Andreas-Forster — CHANGES_REQUESTED** [review]

File: src/main/scala/scalismo/io/stl/STLHelpers.scala [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  "vertex",
+  "vertex",
+  "vertex",
+  "endloop",
+  "endfacet"
+)
+
+case class STLTriangle(n: EuclideanVector3D, p1: Point3D, p2: Point3D, p3: Point3D)
+
+package object STLHelpers {
```
Comment (Andreas-Forster) [inline src/main/scala/scalismo/io/stl/STLHelpers.scala:41]: Why is this a package object? Is there a reason for it?

**Review by Andreas-Forster — APPROVED** [review]

**Pull request merged**